src/metabase/models/notification.clj

```diff
@@ -532,3 +540,15 @@
   "Update an existing notification with `new-notification`."
   [existing-notification new-notification]
   (models.u.spec-update/do-update! existing-notification new-notification notification-update-spec))
+
+(defn unsubscribe-user!
+  "Unsubscribe a user from a notification."
+  [notification-id user-id]
+  (t2/delete! :model/NotificationRecipient
+              {:where [:and
+                       [:= :user_id user-id]
+                       [:exists {:select [1]
```

Comment:
Not a SQL query optimizer expert, but I worry that this approach is really expensive if the user has a lot of subscriptions. Can't you do something like `[:in :notification_recipient.notification_handler_id <subselect>]`?

Reply:
switched to an inner join

Reply:
rewrote to use subselect